src/com/egeniq/widget/SpannableTextView.java

```diff
@@ -0,0 +1,107 @@
+package com.egeniq.widget;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Typeface;
+import android.text.SpannableString;
+import android.text.Spanned;
+import android.text.style.MetricAffectingSpan;
+import android.text.style.TextAppearanceSpan;
+import android.util.AttributeSet;
+
+import com.egeniq.utils.loader.TypefaceLoader;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by josvanegmond on 7/27/15.
+ * Generic class for marking up a TextView with multiple MetricAffectingSpan objects.
+ */
+public class SpannableTextView extends TextView {
+
+    /**
+     * Data class for storing Markup data for each span applied to this TextView.
+     */
+    private class _MarkupData {
```

Comment:
Isn't MarkupData associated with an instance of SpannableTextView instead of the class?